### PR TITLE
APSR-1950: Fix full phone number showing when sending code for 2SV

### PR DESCRIPTION
**Pull request by kskourtis1**

### CODE REVIEW

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/helpers/PhoneNumberHelper.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.thirdpartydeveloperfrontend.helpers
+
+object PhoneNumberHelper {
+
+  def obfuscateNumber(mobileNumber: String): String = mobileNumber.substring(0, mobileNumber.length - 4).replaceAll("\\S", "*") + mobileNumber.takeRight(4)
```
Comment (johnsgp) [inline app/uk/gov/hmrc/thirdpartydeveloperfrontend/helpers/PhoneNumberHelper.scala:21]:
```suggestion
  def obfuscateNumber(mobileNumber: String): String = "*" * 6 + mobileNumber.takeRight(4)
```

This looks overly complicated. Does the user need to know how many digits are in their phone number?

**Comment by johnsgp** [comment]
Is there a case for obfuscating in the back-end (third-party-developer) so that it doesn't have to be done in the front-end. Does the clear text number ever need to be shown in the front-end?

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/helpers/PhoneNumberHelper.scala (hunk 1 above)
Comment (kskourtis1) [reply app/uk/gov/hmrc/thirdpartydeveloperfrontend/helpers/PhoneNumberHelper.scala:21]: On this one I was following the practice on tpd, so I assumed it was all fine
https://github.com/hmrc/third-party-developer/blob/0d448404a445f8daf930e82779ce4b797c46a4dc/app/uk/gov/hmrc/modules/mfa/utils/PhoneNumberHelper.scala#L18

**Comment by kskourtis1** [comment]
As this number was returned was within TpdUser from the main TPD fetch developer call, I presumed that there might be legitimate situations were that number would need to be returned from the call, it didn't seem a good idea to modify the data that the call returns. In my mind it's a frontend concern that the data at that point needs to be masked

**Review by Tobias1087 — APPROVED** [review]

**Pull request merged**